src/hotspot/share/opto/parse1.cpp

```diff
@@ -2393,7 +2396,8 @@ void Parse::return_current(Node* value) {
         inc_sp(1);
         value = value->as_InlineType()->allocate_fields(this);
       }
-    } else if (value->is_InlineType()) {
+    } else if ((is_vector_value && skip_scalarization) ||
+               (value->Opcode() != Op_VectorBox && value->is_InlineType())) {
```

Comment:
`is_vector_value` can be removed in the `else-if` condition since `skip_scalarization = is_vector_value && Compile::current()->inlining_incrementally();` , right?

Reply:
It seems `value->Opcode() != Op_VectorBox` in the second line is not needed, since the VectorBox's klass is vector which is contained in `skip_scalarization`, right?